ravenframework/SupervisedLearning/ARMA.py

```diff
@@ -457,9 +455,17 @@ def __setstate__(self, d):
       @ In, d, dict, stateful dictionary
       @ Out, None
     """
-    rngCounts = d.pop('crow_rng_counts')
+    rngCounts = d.pop('crow_rng_counts', None)
     super().__setstate__(d)
-    self.setEngine(randomUtils.newRNG(), seed=None, count=rngCounts)
+
+    try:
+      self.randomEng
+    except AttributeError:  # catches where ARMA was pickled without saving the RNG
+      self.setEngine(randomUtils.newRNG(), seed=None, count=rngCounts)
+    else:
+      if isinstance(self.randomEng, findCrowModule('randomENG').RandomClass):
+        self.randomEng = randomUtils.RNG(self.randomEng, self.seed)  # wraps raw crow RNG
```

Comment:
I guess you need to change randomUtils.RNG to randomUtils.CrowRNG, could you change it in your PR? @j-bryan

Reply:
In addition, why the count is not settled in the self.randomEng? @j-bryan

Reply:
For the `else` block in question, I think the best way forward would be to get rid of it entirely. I included it to handle an edge case where a raw crow RNG is found in `self.randomEng` when unpickling the ARMA ROM, but that shouldn't be possible. Eliminating that block shouldn't change the functionality and would be the cleanest solution. What are your thoughts @wangcj05?

I also found a mistake in the CrowRNG constructor docstring, so I'll include that correction and some minor cleanup of the constructor code in an update to the other PR.

Reply:
To clarify on the handling of the count, it is handled in the CrowRNG wrapper. What's left in the __setstate__ method is so that existing pickled ROMs could still be loaded even if they didn't store `self.randomEng` when being pickled. Should I remove this, even if that'll cause existing ROM pickles to fail to load?

Reply:
I think I agree with you, when self.randomEng exists, I think there is no need to set it again. 
Regarding the second question, I see your point, the count will be set when we load the engine. My question is mainly for above line where you try to reset the engine, but I think in this line, the count is not set yet. 
@j-bryan

Reply:
@wangcj05 Yes, I think that line 467 should handle the count, if we don't just remove that section entirely. I believe the count is being handled correctly in the other cases.

Reply:
@j-bryan I have two suggestions here:
1. remove the else check
2. make the CrowRNG to a more general wrapper for both Crow and numpy.

Reply:
After talking point 2 over, it might be good to open a discussion on what the advantages a class-based RNG function interface would be over the current utility functions in randomUtils.py. @PaulTalbot-INL Perhaps you have some insights here?